core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailSourceConnection.java

```diff
@@ -1033,4 +1035,10 @@ private boolean hasStatement(SailDataset dataset, Resource subj, IRI pred, Value
 		}
 	}
 
+	protected long calculateSize(final boolean includeInferred, final Resource... contexts) throws SailException {
+		try (SailSource branch = branch(IncludeInferred.fromBoolean(includeInferred))) {
+			return branch.dataset(getIsolationLevel()).size(null, null, null, contexts);
+		}
+	}
+
```

Comment:
Any reason for why we need a new method here?

From a quick glance I saw that there is `protected long sizeInternal(Resource... contexts) throws SailException {` which I wonder if would be a better place for this code?